src/averageAges.js

```diff
@@ -1,61 +1,86 @@
 'use strict';
 
-/**
- * Implement calculateMenAverageAge function
- *
- * Function returns average age of men in array. If `century` is specified then
- * function calculates average age only for men who died in this century
- *
- * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
- *
- * @param {object[]} people
- * @param {number} century - optional
- *
- * @return {number}
- */
+// Methods for working with data base 'people'
+const methods = {
+  getFilteredPeopleBySex(people, sex) {
+    return people.filter(person => person.sex === sex);
+  },
+
+  getFilteredMenByCentury(people, century) {
+    return people.filter(person => Math.ceil(person.died / 100) === century);
+  },
+
+  calculateTotalAge(people) {
+    return people
+      .reduce((ages, person) => ages + (person.died - person.born), 0);
+  },
+
+  getAverage(list) {
+    const totalAge = methods.calculateTotalAge(list);
+
+    return totalAge / list.length;
+  },
+
+  getOnlyMothersList(people) {
+    const onlyMothersList = [];
+
+    people.forEach(person => {
+      if (typeof person.mother === 'string') {
+        onlyMothersList.push(person.mother);
+      }
+    });
+
+    const uniqueMothersList = [...new Set(onlyMothersList)];
+    const mothers = uniqueMothersList.map(motherName => {
+      return people.find(person => person.name === motherName);
+    });
+
+    return mothers.filter(mother => typeof mother !== 'undefined');
+  },
```

Comment:
Overthought, used a lot of code for such a simple filter.
You need only to filter those people that have another person with property name === current person's property mother. It's just a `filter` with `some` methods.